server/controllers/notification.js

```diff
@@ -0,0 +1,176 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const {
+    user,
+    body: { notifyType, title, description, receivers },
+  } = req;
+
+  const sender = await Profile.findOne({ userId: user.id });
+
+  if (!sender) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  const notification = await new Notification({
+    sender: sender.id,
+    notifyType,
+    title,
+    description,
+    receivers,
+  }).save();
+
+  if (notification) {
+    res.status(201).json({
+      success: {
+        notification: {
+          _id: notification.id,
+          notifyType: notification.notifyType,
+          title: notification.title,
+          description: notification.description,
+          sender: {
+            userId: user.id,
+            name: sender.name,
+            photo: sender.photo,
+          },
+          receivers: notification.receivers,
+          readBy: notification.readBy,
+          createdAt: notification.createdAt,
+        },
+      },
+    });
+  } else {
+    res.status(400);
+    throw new Error("Invalid notification data");
+  }
+});
+
+exports.markNotificationRead = asyncHandler(async (req, res, next) => {
+  const {
+    user,
+    body: { id },
+  } = req;
+
+  const receiver = await User.findById(user.id);
+
+  if (!receiver) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  const query = {
+    _id: id,
+    receivers: {
+      $in: receiver.id,
+    },
+  };
+
+  const data = {
+    $set: {
+      readBy: {
+        receiverId: receiver.id,
+      },
+    },
+  };
+
+  const notification = await Notification.findOneAndUpdate(query, data, {
+    new: true,
+  })
+    .populate({
+      path: "sender",
+      select: "userId name photo",
+    })
+    .select("-receivers -__v -updatedAt")
+    .exec();
```

Comment:
This controller is nice overall, but using `.save()` over other update methods will usually lead to simpler implementations. It's also recomended by [mongoose](https://mongoosejs.com/docs/documents.html#updating-using-save): 
> The save() function is generally the right way to update a document with Mongoose. With save(), you get full validation and middleware.